src/state/reducer.ts

```diff
@@ -13,6 +13,14 @@ type Writable<T> = {
   [P in keyof T]: NonNullable<T[P]>;
 }
 
+function resourceCustomizer(a, b) {
+	if (b instanceof Resource) {
+			const merged = mergeWith({ ...a }, b, resourceCustomizer);
+
+			return Object.assign(b, merged);
```

Comment:
The source Resource instance is mutated, which in this case will be from `normalize()`, and should be safe to mutate. There is probably a better way to do this.

Reply:
any mutation of existing state in a reducer will have bad side effects. You'll need to copy on merge. Perhaps adding a new bulk update method to Resource could be of assistance.

Reply:
I agree. This mutation felt very dirty, and even though it might be safe right now, it is likely to cause bugs that are difficult to debug if that changes.

I figured the better option was to add a new method on Resource, but didn't want to spend to much time on this until I got some feedback.

Reply:
Probably something resembling https://immutable-js.github.io/immutable-js/docs/#/Map/merge since I'll want to eventually have this library optionally use immutable-js